internal/metricagent/collect_metric.go

```diff
@@ -0,0 +1,118 @@
+package metricagent
+
+import (
+	"context"
+	"runtime"
+	"sync/atomic"
+	"time"
+
+	"github.com/andreevym/metric-collector/internal/logger"
+	"github.com/andreevym/metric-collector/internal/storage"
+	"github.com/shirou/gopsutil/v3/cpu"
+	"github.com/shirou/gopsutil/v3/mem"
+	"go.uber.org/zap"
+)
+
+func collectMetric(
+	ctx context.Context,
+	pollDuration time.Duration,
+	rateLimit int,
+) (chan []*storage.Metric, error) {
+	// создаем буферизованный канал для отправки результатов
+	metricsCh := make(chan []*storage.Metric, rateLimit)
+
+	go func() {
+		defer close(metricsCh)
+
+		// pollCountAtomic (тип counter) — счётчик, увеличивающийся на 1
+		// при каждом обновлении метрики из пакета runtime (на каждый pollInterval — см. ниже).
+		var pollCountAtomic atomic.Int64
+		ticker := time.NewTicker(pollDuration)
+		for t := range ticker.C {
+			metrics := make([]*storage.Metric, 0)
```

Comment:
```suggestion
			metrics := make([]storage.Metric, 0, metricsNum)
```
слайс - это уже ссылочный тип, нет смысла внутри хранить ссылки. Если значения нет - его туда можно просто не записывать. Второе - слайс нужно стараться создавать с заданным капасити чтобы избежать лишних расширений, тут мы это можем сделать т.к. знаем точное число метрик